packages/back-end/src/api/experiments/putExperiment.ts

```diff
@@ -0,0 +1,34 @@
+import { ExperimentInterface } from "../../../types/experiment";
+import { PutExperimentResponse } from "../../../types/openapi";
+import { createExperiment } from "../../models/ExperimentModel";
+import { toExperimentApiInterface } from "../../services/experiments";
+import { createApiRequestHandler } from "../../util/handler";
+import { putExperimentValidator } from "../../validators/openapi";
+
+export const putExperiment = createApiRequestHandler(putExperimentValidator)(
+  async (req): Promise<PutExperimentResponse> => {
+    const newExperiment: Partial<ExperimentInterface> = {
+      ...req.body,
+      dateCreated: new Date(),
+      dateUpdated: new Date(),
+      datasource: req.body.datasourceId,
+      organization: req.organization.id,
+      implementation: "code",
+      phases: [],
```

Comment:
A few comments:
1. `dateCreated`, `dateUpdated`, and `organization` are not used.  They are set explicitly inside of `createExperiment` below.
2. By setting `phases` here we are overwriting what was passed into the request body.  If we want to make this empty array a default instead of an override, we can move it before the `...req.body` spread operator

Reply:
All good points, fixed.

For the `dateStarted` field within `phases`, I get this typescript error if moving `phases` before `...req.body`. Not TS expert I am, could you help me resolve this?

<img width="864" alt="image" src="https://github.com/growthbook/growthbook/assets/13710518/564a1d4b-f88d-4c18-8fa9-a3333adb27a0">